src/Core/Resolvers/SqlMutationEngine.cs

```diff
@@ -923,6 +956,738 @@ private FindRequestContext ConstructFindRequestContext(RestRequestContext contex
+
+                if (dbResultSetRowForCurrentEntity is not null && dbResultSetRowForCurrentEntity.Columns.Count == 0)
+                {
+                    // For GraphQL, insert operation corresponds to Create action.
+                    throw new DataApiBuilderException(message: $"Could not insert row with given values for entity: {entityName}",
+                                                      statusCode: HttpStatusCode.Forbidden,
+                                                      subStatusCode: DataApiBuilderException.SubStatusCodes.DatabasePolicyFailure);
+                }
+
+                if (dbResultSetRowForCurrentEntity is null)
+                {
+                    throw new DataApiBuilderException(
+                        message: "No data returned back from database.",
+                        statusCode: HttpStatusCode.InternalServerError,
+                        subStatusCode: DataApiBuilderException.SubStatusCodes.DatabaseOperationFailed);
+                }
+
+                Dictionary<string, object?> insertedValues = dbResultSetRowForCurrentEntity.Columns;
+                Dictionary<string, object?> pkValues = new();
+                foreach (string pk in primaryKeyColumnNames)
+                {
+                    pkValues.Add(pk, insertedValues[pk]);
+                }
+
+                primaryKeysOfCreatedItem.Add(entityName, pkValues);
+                multipleCreateStructure.CurrentEntityPKs = pkValues;
+
+                //Perform an insertion in the linking table if required
+                if (multipleCreateStructure.IsLinkingTableInsertionRequired)
+                {
+                    if (multipleCreateStructure.LinkingTableParams is null)
+                    {
+                        multipleCreateStructure.LinkingTableParams = new Dictionary<string, object?>();
+                    }
+
+                    // Add higher level entity PKs
+                    List<ForeignKeyDefinition> foreignKeyDefinitions = relationshipData!.TargetEntityToFkDefinitionMap[multipleCreateStructure.HigherLevelEntityName];
+                    ForeignKeyDefinition fkDefinition = foreignKeyDefinitions[0];
+
+                    int countOfReferencingColumns = fkDefinition.ReferencingColumns.Count;
+                    for (int i = 0; i < countOfReferencingColumns; i++)
+                    {
+                        string referencingColumnName = fkDefinition.ReferencingColumns[i];
+                        string referencedColumnName = fkDefinition.ReferencedColumns[i];
+
+                        if (multipleCreateStructure.LinkingTableParams.ContainsKey(referencingColumnName))
+                        {
+                            continue;
+                        }
+
+                        multipleCreateStructure.LinkingTableParams.Add(referencingColumnName, multipleCreateStructure.CurrentEntityPKs![referencedColumnName]);
```

Comment:
This logic to loop through the FK definition and add values for all the relationship fields from the source/target entity to the linking table params is identical for the two loops here. Can be extracted out into a helper method.

Reply:
They are not identical. Please note that in the first loop, `multipleCreateStructure.LinkingTableParams.Add(referencingColumnName, multipleCreateStructure.CurrentEntityPKs![referencedColumnName]);` is performed.

In the second loop, `multipleCreateStructure.LinkingTableParams.Add(referencingColumnName, multipleCreateStructure.HigherLevelEntityPKs![referencedColumnName]);` is performed.

Reply:
That's what. To the helper, you can just pass either `HigherLevelEntityPKs` or `CurrentEntityPKs`. Rest of the logic is same.

Reply:
Updated